### PR TITLE
feat: extend kadclient to up-/download files

### LINKED ISSUE #100: Have client upload files instead of chunks.
Use the files api, and make sure it is encapsulated.

### PR DESCRIPTION
**Pull request by oetyng**
Resolves #100.

It can now upload and download entire files, instead of small chunks.

Additionally, the files apis are encapsulated in their own struct, as
to not bloat the client api.

### CODE REVIEW

File: safenode/src/bin/kadclient.rs [hunk 1]
```diff
@@ -101,11 +98,11 @@ async fn main() -> Result<()> {
         }
 
         for xorname in chunks_to_fetch.iter() {
-            println!("Fetching chunk {xorname:?}");
-            match client.get_chunk(ChunkAddress::new(*xorname)).await {
-                Ok(chunk) => info!("Successfully got chunk {}!", chunk.name()),
+            println!("Downloading file {xorname:?}");
```
Comment (maqi) [inline safenode/src/bin/kadclient.rs:101]: this is a chunk downloaded, not the whole file

File: safenode/src/bin/kadclient.rs [hunk 2]
```diff
@@ -101,11 +98,11 @@ async fn main() -> Result<()> {
...
+            match file_api.read_bytes(ChunkAddress::new(*xorname)).await {
+                Ok(bytes) => info!("Successfully got file {xorname} of {} bytes!", bytes.len()),
```
Comment (maqi) [inline safenode/src/bin/kadclient.rs:103]: same comment as above, this is just a chunk part of the file fetched

File: safenode/src/bin/kadclient.rs (hunk 1 above)
Comment (oetyng) [reply safenode/src/bin/kadclient.rs:101]: No, @maqi , check the innards of `read_bytes`. It gives a whole file.

It takes a chunk name, checks if it is a head chunk asked for (the one holding the data map, or the final one if data map chunkling resulted in more than one chunk).

**Review by grumbach — APPROVED** [review]
Nice one, files are a much better API than Chunks!
LGTM

**Pull request merged**